### PR TITLE
fix(evals): remove agent behavior warnings and validate environment paths

### LINKED ISSUE #2974: fix(evals): remove agent behavior warnings and validate environment source paths
The agent behavior evaluation suite emits avoidable static warnings from legacy environment keys and custom judges without explicit rubrics. Environment source validation is also inconsistent between root and stimulus scopes, including alias conflicts and path resolution.

This affects contributors maintaining generated evaluation specs because valid absolute paths can be resolved incorrectly, invalid paths are not diagnosed consistently, and warning output obscures actionable validation failures.

## Steps to Reproduce

1. Generate or validate `evals/agent-behavior/eval.yaml` from its canonical stimulus partials.
2. Run strict Vally lint and the shared eval schema validator.
3. Observe the legacy-key deprecation and custom-judge diagnostics.
4. Exercise root-level and stimulus-level environment declarations with named references, relative paths, absolute paths, conflicting aliases, and missing files.

## Expected Behavior

The canonical suite generates without deprecation or rubric diagnostics. Root-level and stimulus-level environment declarations follow the same alias and path-validation rules without modifying valid absolute paths.

## Acceptance Criteria

- [ ] Canonical agent behavior stimulus partials use the supported environment key without changing normalized environment values.
- [ ] The generated agent behavior spec is synchronized with its canonical partials and emits no Vally deprecation or rubric diagnostics.
- [ ] Each affected custom judge has explicit, artifact-grounded rubric criteria while preserving its prompt, grader, threshold, and other evaluation behavior.
- [ ] Root-level and stimulus-level environment declarations accept both supported aliases and named environment references.
- [ ] Relative paths resolve from the declaring spec directory, while valid absolute paths remain unchanged.
- [ ] Conflicting aliases and missing environment files produce structured validation errors without throwing.
- [ ] Schema and generator regression tests cover the repaired behavior and pass.

## Out of Scope

- Live model or moderation evaluation
- Dependency, workflow, public API, or distributable Copilot artifact changes

### PR DESCRIPTION
**Pull request by jkim323**
# Pull Request

## Description

Removes static warnings from the agent behavior evaluation suite while preserving normalized environment data and existing generated-spec behavior.

This change:

* Migrates legacy environment keys in canonical stimulus partials and regenerates the consolidated agent behavior spec
* Adds explicit rubric criteria for five custom-judge evaluations so their expected behavior is grounded in the evaluated artifacts
* Applies shared environment validation at root and stimulus scope, including alias conflicts, named references, relative paths, absolute paths, and structured invalid-path diagnostics
* Expands schema and generator regression coverage, including preservation of absolute source paths

## Related Issue(s)

Fixes #2974 

## Type of Change

Select all that apply:

**Code & Documentation:**

* [x] Bug fix (non-breaking change fixing an issue)
* [ ] New feature (non-breaking change adding functionality)
* [ ] Breaking change (fix or feature causing existing functionality to change)
* [ ] Documentation update

**Infrastructure & Configuration:**

* [ ] GitHub Actions workflow
* [ ] Linting configuration (markdown, PowerShell, etc.)
* [ ] Security configuration
* [ ] DevContainer configuration
* [ ] Dependency update

**AI Artifacts:**

* [ ] Reviewed contribution with `hve-builder` and addressed all actionable findings
* [ ] Copilot instructions (`.github/instructions/*.instructions.md`)
* [ ] Copilot prompt (`.github/prompts/*.prompt.md`)
* [ ] Copilot agent (`.github/agents/*.agent.md`)
* [ ] Copilot skill (`.github/skills/*/SKILL.md`)
* [ ] Copilot hook (`.github/hooks/*/*.json`)
* [x] Eval spec added/updated for changed AI artifacts (`evals/`)

> Note for AI Artifact Contributors:
>
> * Agents: Research, indexing/referencing other project (using standard VS Code GitHub Copilot/MCP tools), planning, and general implementation agents likely already exist. Review `.github/agents/` before creating new ones.
> * Skills: Must include both bash and PowerShell scripts. See [Skills](../docs/contributing/skills.md).
> * Model Versions: Contributions **MUST** target models listed in the model catalog (`scripts/linting/model-catalog.json`) whose provider appears in `providerAllowlist` and whose status is `ga` or `preview`. Run `npm run lint:models` to validate references.
> * See [Agents Not Accepted](../docs/contributing/custom-agents.md#agents-not-accepted) and [Model Version Requirements](../docs/contributing/ai-artifacts-common.md#model-version-requirements).

**Other:**

* [x] Script/automation (`.ps1`, `.sh`, `.py`)
* [ ] Other (please describe):

## Sample Prompts (for AI Artifact Contributions)

**User Request:**

**Execution Flow:**

**Output Artifacts:**

**Success Indicators:**

For detailed contribution requirements, see:

* Common Standards: [docs/contributing/ai-artifacts-common.md](../docs/contributing/ai-artifacts-common.md) - Shared standards for XML blocks, markdown quality, RFC 2119, validation, and testing
* Agents: [docs/contributing/custom-agents.md](../docs/contributing/custom-agents.md) - Agent configurations with tools and behavior patterns
* Prompts: [docs/contributing/prompts.md](../docs/contributing/prompts.md) - Workflow-specific guidance with template variables
* Instructions: [docs/contributing/instructions.md](../docs/contributing/instructions.md) - Technology-specific standards with glob patterns
* Skills: [docs/contributing/skills.md](../docs/contributing/skills.md) - Task execution utilities with cross-platform scripts

## Testing

Passed:

* `git diff --check origin/main...HEAD`
* `pwsh -NoProfile -File scripts/evals/Build-AgentBehaviorSpec.ps1 -WhatIf`
* `npm run lint:eval-grader-lineage`
* `npm exec -- vally lint --eval-spec evals/agent-behavior/eval.yaml --strict`
* `pwsh -NoProfile -File scripts/evals/Test-EvalSpec.ps1`
* `npm run test:ps -- -TestPath 'scripts/tests/evals/Test-EvalSpec.Tests.ps1'` (49 passed)
* `npm run test:ps -- -TestPath 'scripts/tests/evals/Build-AgentBehaviorSpec.Tests.ps1'` (75 passed, one Integration-tagged test not run by the default runner)
* `npm run lint:ps`

### Required Checks

* [ ] Documentation is updated (if applicable)
* [x] Files follow existing naming conventions
* [x] Changes are backwards compatible (if applicable)
* [x] Tests added for new functionality (if applicable)

### AI Artifact Contributions

* [ ] Used `hve-builder` review mode to review contribution
* [ ] Addressed all actionable findings from the `hve-builder` review
* [ ] Verified contribution follows common standards and type-specific requirements

### Required Local Checks

The following local-safe validation commands must pass before merging:

* [x] Local validation aggregate: `npm run validate:local`
* [x] Documentation validation (if docs changed): `npm run validate:docs`
* [x] Spell checking: `npm run spell-check`
* [x] Link validation: `npm run lint:md-links`

## Security Considerations

* [x] This PR does not contain any sensitive or NDA information
* [x] Any new dependencies have been reviewed for security issues
* [x] Security-related scripts follow the principle of least privilege

## Additional Notes

* Static validation reports no current Vally warnings or diagnostics for the changed suite
* Live model evaluation, content moderation, and hosted CI checks were not run locally and remain pending in CI
* No dependencies, workflows, public APIs, or distributable Copilot artifacts changed

### CODE REVIEW

File: scripts/evals/Modules/EvalSpecSchema.psm1 [hunk 1]
```diff
@@ -85,6 +85,100 @@ function Resolve-EvalArtifactPath {
...
+            }
+            else {
+                [string]$rawPath
+            }
+            if ([string]::IsNullOrWhiteSpace($pathString)) {
+                $errors.Add(@{ path = $SpecPath; field = $entryField; message = "Empty $field.$entryKey path" })
+                continue
+            }
+            try {
+                $resolved = [System.IO.Path]::GetFullPath($pathString, $SpecDirectory)
```
Comment (WilliamBerryiii) [inline scripts/evals/Modules/EvalSpecSchema.psm1:172]: Thanks for centralizing this validation. Could we reject UNC and device paths before this filesystem probe? On Windows, `Test-Path` against an eval-controlled UNC source can initiate SMB authentication during static validation. Please preserve valid local absolute paths and add a regression test showing rejected network paths never reach `Test-Path`.

**Review by WilliamBerryiii — CHANGES_REQUESTED** [review]
Thank you for the focused eval cleanup and regression coverage. I left one inline security concern about network paths reaching `Test-Path`; addressing it while preserving valid local absolute-path support should resolve this request.

File: scripts/evals/Modules/EvalSpecSchema.psm1 (hunk 1 above)
Comment (jkim323) [reply scripts/evals/Modules/EvalSpecSchema.psm1:172]: Addressed—UNC and device paths are now rejected before Test-Path, with regression coverage confirming no filesystem probe occurs. Valid local absolute paths remain supported.

**Review by chaosdinosaur — COMMENTED** [review]
Thank you for the focused eval cleanup and regression coverage. The functional
and standards passes found the canonical and generated artifacts synchronized,
and the targeted Pester suites pass. I found one source-boundary concern: the
new lexical network/device guard does not establish
that an accepted resolved source is an ordinary local file or directory after
link resolution. Please enforce the intended source policy at the actual
staging boundary and add platform-specific regression coverage. Before merge,
also update from main, obtain a fresh approval, and disposition the hosted
advisory assertions before merge.

**Review by chaosdinosaur — APPROVED** [review]

File: evals/agent-behavior/eval.yaml
Comment (rezatnoMsirhC) [inline evals/agent-behavior/eval.yaml:1]: Thanks for tightening the environment validation. After the recent merges from main, the agent-behavior suite could be warning-free again once a few more partials are migrated. #2914 and #2915 added 18 stimulus-level `environment:` keys in `accessibility-framework-assessor.yml`, `accessibility-planner.yml`, `accessibility-reviewer.yml`, `accessibility-surface-inventory.yml`, and `data-science-engineering-coach.yml`, and they flow into the generated `eval.yaml`. The latest Eval Validation run on this head still logs `DeprecationWarning: eval spec: 'environment' is deprecated, use 'agent_environment' instead` immediately after `evals/agent-behavior/eval.yaml is valid` ([job log](https://github.com/microsoft/hve-core/actions/runs/35939594396/job/107444328763)), so the "emits no Vally deprecation" criterion in #2974 isn't met yet. Could we rename those keys to `agent_environment` in the partials and regenerate with `scripts/evals/Build-AgentBehaviorSpec.ps1`? It might also help to have a lightweight guard that fails when the old key appears under `evals/agent-behavior/`, so later merges don't bring it back.

Comment (jkim323) [reply evals/agent-behavior/eval.yaml:1]: Thanks for the catch. I addressed it by removing the static warnings from the agent behavior evals while preserving the normalized environment data and existing generated-spec behavior. I also added the missing validation at the root and stimulus scope, covered the alias/path edge cases, and updated the schema/regression tests to lock in the expected behavior

**Review by rezatnoMsirhC — APPROVED** [review]
